src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -273,14 +287,42 @@ private static void TryLoadGeneratedAssemblyFromCache(Assembly targetAssembly)
         private static CachedAssembly CompileAndLoad(GeneratedSyntax generatedSyntax, bool emitDebugSymbols)
         {
             var generated = CodeGeneratorCommon.CompileAssembly(generatedSyntax, "OrleansCodeGen", emitDebugSymbols: emitDebugSymbols);
-            Assembly.Load(generated.RawBytes, generated.DebugSymbolRawBytes);
+            var loadedAssembly = LoadAssembly(generated);
             return new CachedAssembly(generated)
             {
-                Loaded = true
+                Loaded = true,
+                Assembly = loadedAssembly,
             };
         }
 
         /// <summary>
+        /// Loads the specified assembly.
+        /// </summary>
+        /// <param name="asm">The assembly to load.</param>
+        private static Assembly LoadAssembly(GeneratedAssembly asm)
+        {
+#if ORLEANS_BOOTSTRAP
+            throw new NotImplementedException();
+#elif NETSTANDARD
+            Assembly result;
+            result = Orleans.PlatformServices.PlatformAssemblyLoader.LoadFromBytes(asm.RawBytes, asm.DebugSymbolRawBytes);
+            AppDomain.CurrentDomain.AddAssembly(result);
+            return result;
+#else
```

Comment:
This assembly is `netstandard1.5`, but it calls into the multi-targeted `PlatformServices` assembly. I didn't want to mess with the existing .NET 4.6 tree (only vNext tree), so we use `Assembly.Load()` directly in the main tree.